cassandra-scheduler/src/main/java/com/mesosphere/dcos/cassandra/scheduler/CassandraScheduler.java

```diff
@@ -246,6 +258,15 @@ public void statusUpdate(SchedulerDriver driver, Protos.TaskStatus status) {
         } catch (Exception ex) {
             LOGGER.error("Error updating Stage Manager with status: {} reason: {}", status, ex);
         }
+
+        // We don't necessarily need to revive on every status update, but we might, and it's
+        // safe to do so.
+        reviveOffers();
```

Comment:
I agree with the comment above. But, is there any reason why we cannot implement `shouldRevive()` by inspecting current state ?